src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpRequestCallback.cs

```diff
@@ -149,7 +149,24 @@ private static void OnRequestReadComplete(WinHttpRequestState state, uint bytesR
             Debug.Assert(state.TcsReadFromResponseStream != null, "TcsReadFromResponseStream is null");
             Debug.Assert(!state.TcsReadFromResponseStream.Task.IsCompleted, "TcsReadFromResponseStream.Task is completed");
             
-            state.TcsReadFromResponseStream.TrySetResult((int)bytesRead);
+            state.CurrentBytesRead += (long)bytesRead;
```

Comment:
Nevermind, that doesn't make sense.  I just realized what the conditions for the if/else are.